tests/integration/mibs/memory_test.py

```diff
@@ -0,0 +1,44 @@
+import pytest
+import pytest_twisted
+
+from nav.mibs import cisco_memory_pool_mib, juniper_mib, netswitch_mib
+
+
+@pytest.mark.twisted
+@pytest_twisted.inlineCallbacks
+def test_juniper_get_memory_usage(snmp_agent_proxy):
+    snmp_agent_proxy.community = 'juniper-memory'
+    snmp_agent_proxy.open()
+    mib = juniper_mib.JuniperMib(snmp_agent_proxy)
+    res = yield mib.get_memory_usage()
+    assert res == {
+        '"FPC: EX2200-48T-4G @ 0/*/*"': (257698037.76, 279172874.24),
+        '"FPC: EX2200-48T-4G @ 1/*/*"': (204010946.56, 332859965.44),
+        '"FPC: EX2200-48T-4G @ 2/*/*"': (182536110.08, 354334801.91999996),
+        '"Routing Engine 0"': (257698037.76, 279172874.24),
+        '"Routing Engine 1"': (204010946.56, 332859965.44),
+    }
```

Comment:
Ooh, this is dangerous. Comparisons involving exact floating point values are never a good idea. Floating point numbers are, by the very nature of how they are represented in a computer, never exact to the decimal. These tests happen to pass right now, in the environment in which they are being run, but they may fail on another computer or in a different version of Python.

There are about a gazillion articles on the net about this type of issue, here just one of them: https://bitbashing.io/comparing-floats.html

I believe pytest provides its own floating point number comparison function, which should probably be used in this case: `pytest.approx`

Reply:
Oh yeah, I remember learning about this, what a classic mistake :woman_facepalming:

Reply:
Yeah, it saddens me that the test code gets unnecessarily ugly because of this - but it's the only safe way.